libp2p/autotls/acme.nim

```diff
@@ -0,0 +1,308 @@
+import options, json, base64
+from times import DateTime, parse
+import chronos/apps/http/httpclient, jwt, results, bearssl/pem
+
+import ./utils
+import ../crypto/crypto
+import ../crypto/rsa
+import ../transports/tls/certificate_ffi
+import ../transports/tls/certificate
+
+# TODO: change staging to actual url
+const
+  LetsEncryptURL* = "https://acme-v02.api.letsencrypt.org"
+  LetsEncryptURLStaging* = "https://acme-staging-v02.api.letsencrypt.org"
+  Alg = "RS256"
+
+type ACMEAccount* = object
+  status*: Opt[string]
+  contact*: Opt[seq[string]]
+  key*: KeyPair
+  session*: HttpSessionRef
+  kid*: Opt[string]
+  directory: JsonNode
+  acmeServerURL: string
+
+proc new*(
+    T: typedesc[ACMEAccount],
+    key: KeyPair,
+    status: Opt[string] = Opt.none(string),
+    contact: Opt[seq[string]] = Opt.none(seq[string]),
+    kid: Opt[string] = Opt.none(string),
```

Comment:
Does it make sense for these fields to be setup when instantiating ACMEAccount? It seems that `status` and `kid` are setup later. Is `contact` being used at all?

Reply:
`contact` is there for future use, we're not using it as of now